src/fluree/ledger/core.clj

```diff
@@ -384,6 +391,139 @@
+            (when (not= 1 (count checksums-with-max))
+              (throw (ex-info (str "Ledger: " ledger " block: " block " has multiple checksum conflicts: " ledger-checksums
+                                   "Unable to resolve!") {})))
+            ;; found a common checksum, copying
+            (println ledger "block" block "checksum-match with" most-common-n "ledgers, copying block from:" (first ledgers-with-checksum))
+            (copy-block ledger (first ledgers-with-checksum) repair-dir block)
+            (ledger-stats-update ledgers-with-checksum)
+            (ledger-stats-errors block :checksum-error (filter #(not (ledgers-with-checksum %)) data-dirs)))
+
+          ;; block 't' value likely missing, see if a ledger has no warning for this block
+          (:warnings block-info)
+          (let [ledger-warning-info      (:warnings block-info)
+                ledger-with-warnings     (into #{} (keys ledger-warning-info))
+                ledgers-without-warnings (filter #(not (ledger-with-warnings %1)) data-dirs)]
+            (when (empty? ledgers-without-warnings)
+              (throw (ex-info (str "Ledger: " ledger " block: " block " has no ledgers without issues: " ledger-warning-info
+                                   "Unable to resolve!") {})))
+            (println ledger "block" block "warnings for all but" (count ledgers-without-warnings) "ledgers, copying block from:"
+                     (first ledgers-without-warnings) "Warnings:" ledger-warning-info)
+            (copy-block ledger (first ledgers-without-warnings) repair-dir block)
+            (ledger-stats-update ledgers-without-warnings)
+            (doseq [ledger ledger-with-warnings]
+              (ledger-stats-errors block (get ledger-warning-info ledger) [ledger])))
+
+          ;; block is missing in one or more ledgers
+          (:missing-in block-info)
+          (let [missing-in     (:missing-in block-info)
+                not-missing-in (filter #(not (missing-in %)) data-dirs)]
+            (when (empty? not-missing-in)
+              (throw (ex-info (str "Ledger: " ledger " block: " block " has no ledgers with block: " block
+                                   "Unable to resolve!") {})))
+            (println ledger "block" block "missing in ledgers" missing-in ". Copying from:" (first not-missing-in))
+            (copy-block ledger (first not-missing-in) repair-dir block)
+            (ledger-stats-update not-missing-in)
+            (ledger-stats-errors block :missing missing-in))
+
+
+          :else
+          (throw (ex-info (str "Ledger: " ledger " block: " block " Unexpected Error!: " block-info) {})))))
+    (let [good-blocks  (get @ledger-stats :good)
+          good-ledgers (filter #(= max-block (get good-blocks %1)) data-dirs)]
+      (if (empty? good-ledgers)
+        (println ">>>> No Ledger was complete!!!!")
+        (println ">>>> Ledger(s)" good-ledgers "were complete!"))
+      @ledger-stats)))
```

Comment:
This fn is pretty long. Are there any decent opportunities to extract some smaller fns out of it?

Reply:
Probably, not something that gets used often and under time crunch, otherwise I would have chunked it up more.